02_bubble/brnbke/main.go

```diff
@@ -0,0 +1,91 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+)
+
+var out io.Writer = os.Stdout
+
+func bubble(arr []int) []int {
+	changed := true
+	r := make([]int, len(arr))
+	copy(r, arr)
+
+	for {
+		if !changed {
+			break
+		}
+		changed = false
+		for i := 0; i < len(r)-1; i++ {
+			if r[i] > r[i+1] {
+				r[i], r[i+1] = r[i+1], r[i]
+				changed = true
+			}
+		}
+	}
+
+	return r
+}
+
+func insertion(arr []int) []int {
+	result := make([]int, len(arr))
+	copy(result, arr)
+
+	for i := 1; i < len(result); i++ {
+		tmp := result[i]
+		for j := i - 1; j > -1; j-- {
+			if result[j] > tmp {
+				result[j+1], result[j] = result[j], tmp
+			} else {
+				break
```

Comment:
nitpick: I'd suggest only indenting the `break` block, and having the other one outside the `if` statement. https://github.com/golang/go/wiki/CodeReviewComments#indent-error-flow

Reply:
I agree with Sam. By using:
```
if result[j] <= tmp {
  break
}
result[j+1], result[j] = result[j], tmp
```
It makes it clear what the terminating condition is - I don't need to mentally invert the condition in my head to understand the condition of the special case.

It is also a variation of aligning the happy path to the left edge: https://medium.com/@matryer/line-of-sight-in-code-186dd7cdea88 (modulo loop indentation).